nonmoodlefeedback.php

```diff
@@ -38,10 +37,16 @@
 $instance = optional_param('instance', 0, PARAM_INT);
 
 $report = new \report_myfeedback\local\report();
-$canaddnotes = $report->can_add_non_moodle_feedback($gradeid, $userid);
+// Check if the current user can add non-Moodle feedback for this user and
+// grade item.
+$gradeitem = $DB->get_record('grade_items', ['id' => $gradeid], '*', MUST_EXIST);
+$coursecontext = context_course::instance($gradeitem->courseid);
+require_capability('report/myfeedback:addnonfeedback', $coursecontext);
 
-if ($canaddnotes !== true) {
-    throw new moodle_exception($canaddnotes, 'report_myfeedback');
+if (!is_enrolled($coursecontext, $USER->id)) {
+    throw new moodle_exception('teachernopermission', 'report_myfeedback');
```

Comment:
Actually, do we need this first condition? Isn't it implicit in the `require_capability()` that the teacher is enrolled on the course (or, at least, has capability `report/myfeedback:addnonfeedback` in this course)? In which case this just needs to be:

```
if (!is_enrolled($coursecontext, $userid)) {
    throw new moodle_exception('studentnotincourse', 'report_myfeedback');
}
```

Reply:
You're right, removed the first condition, thanks for the review, really helpful!